libmcount/dynamic.c

```diff
@@ -315,7 +316,7 @@ static int find_dynamic_module(struct dl_phdr_info *info, size_t sz, void *data)
 		free(mdi);
 	}
 
-	return !fmd->needs_modules;
+	return !fmd->needs_modules && is_executable;
```

Comment:
On Android linker64 precedes the executable in list of modules so we can not bail out after the first module.